packages/block-editor/src/hooks/position.js

```diff
@@ -214,40 +270,58 @@ export function PositionPanel( props ) {
 	const {
 		attributes: { style = {} },
 		clientId,
+		isSelected,
 		name: blockName,
 		setAttributes,
 	} = props;
 
 	const allowFixed = hasFixedPositionSupport( blockName );
 	const allowSticky = hasStickyPositionSupport( blockName );
 	const value = style?.position?.type;
+	const [ isPositionVisualizerActive, setIsPositionVisualizerActive ] =
+		useState();
 
-	const { hasParents } = useSelect(
+	const { firstParentClientId } = useSelect(
 		( select ) => {
 			const { getBlockParents } = select( blockEditorStore );
 			const parents = getBlockParents( clientId );
-			return {
-				hasParents: parents.length,
-			};
+			return { firstParentClientId: parents[ parents.length - 1 ] };
 		},
 		[ clientId ]
 	);
 
+	const onMouseOverPosition = () => {
+		setIsPositionVisualizerActive( true );
+	};
+
+	const onMouseLeavePosition = () => {
+		setIsPositionVisualizerActive( false );
+	};
+
+	const blockInformation = useBlockDisplayInformation( firstParentClientId );
+	const stickyHelpText =
+		allowSticky && value === STICKY_OPTION.value && blockInformation
+			? sprintf(
+					/* translators: %s: the name of the parent block. */
+					__(
+						'The block will stick to the scrollable area of the parent %s block.'
```

Comment:
Not a blocker, just asking:

Is there a way to tell if a block is nested inside a template? 

I know [getting sticky to work on templates is off the cards for now](https://github.com/WordPress/gutenberg/pull/49321#issuecomment-1524407965), but I guess folks might try to make the header sticky and until we can get that working this doesn't make much sense when I try to make Header > Group sticky:

> The block will stick to the scrollable area of the parent Header block.

Reply:
> Is there a way to tell if a block is nested inside a template?

Good question!

Unfortunately it's a tricky one to figure out, because depending on what a user's site layout is like, there could be a good case for a user making their block sticky to the immediate parent within a header. I suppose it's one of the challenges of exposing the controls in non-root positions — we'd like to try to help communicate what's going to happen, but at the same time, it's up to users to use the tools however it suits their current designs 🤔

Personally, I'd probably explore tweaking logic surrounding where the block sits in relation to templates / template parts in follow-ups, as it gets quite tricky to come up with something the feels just right.

Thanks for taking a look!